File: features/dd-sdk-android-ndk/src/main/cpp/utils/file-utils.cpp
/*
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2016-Present Datadog, Inc.
 */

#include "file-utils.h"

#include <cerrno>
#include <dirent.h>
#include <sys/stat.h>

namespace fileutils {

    bool create_dir_if_not_exists(const char *dirPath) {
        if (opendir(dirPath) == nullptr && ENOENT == errno) {
            // directory does not exist. We will create it.
            return mkdir(dirPath, S_IRWXU) == 0;
        }

        return true; // the directory was already there
    }
}

File: features/dd-trace-core/src/main/java/com/datadog/trace/util/FileUtils.java
/*
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2016-Present Datadog, Inc.
 */

package com.datadog.trace.util;

import java.io.ByteArrayOutputStream;
import java.io.FileInputStream;
import java.io.IOException;

public class FileUtils {

    public static byte[] readAllBytes(String path) throws IOException {
        try (FileInputStream inputStream = new FileInputStream(path)) {
            ByteArrayOutputStream outputStream = new ByteArrayOutputStream();
            int bytesRead;
            byte[] buffer = new byte[1024];
            while ((bytesRead = inputStream.read(buffer, 0, buffer.length)) != -1) {
                outputStream.write(buffer, 0, bytesRead);
            }
            outputStream.flush();
            return outputStream.toByteArray();
        }
    }

}
